crawler/rss.py

```diff
@@ -16,26 +16,37 @@ def preprocess(cls, source, options):
         return [RSSConfig.links[source]]
     
     @classmethod
-    def extract(cls, objs, options):
+    def extract(cls, objs, code, options):
         """
         Extract news from BeautifulSoup object.
         """
         # TODO Optimize to each newspaper style
         soup = objs[0]
-        ret = list()
-        for entry in soup.findAll("entry"):
+        ret = list()      
+
+        for entry in soup.findAll(RSSConfig.item_name[code]):
             temp = dict()
+            temp["category"] = list()
             for item in entry.children:
-                # print(item.name)
-                if item.name == None: 
+                if item.name == None:   
```

Comment:
We can merge this condition with the condition on line 33:

if item.name == None or item.name == RSSConfig.item_publish[code]: